### ISSUE TITLE
Hi, I want to known about "ProtocolBuffers\Message::getExtension(string $name)"

### ISSUE THREAD

**Issue by willsky** [chobie/php-protocolbuffers#42]
what's about this extentsion! Could you explain to me! Thank you!

**Comment by willsky** [comment]
And use case!

**Comment by chobie** [comment]
Sorry, I've just missed this issue.  please check this docs.
https://developers.google.com/protocol-buffers/docs/proto#extensions

let assume some vendor or platform provides greeting.proto like this.

```
message Greeting{
  optional id int32 = 1;
  optional bytes message = 2;
  extensions 100 to 199;
}
```

The message reserved extension which number in 100 to 199. you can custom this tag number without change original message file.

```
extend Greeting {
  optional int32 language_id = 100;
}
``

re-generate files. then you can use 100 tag as language_id.

for example. protoc-gen-php uses extension for parsing php specific options.
https://github.com/chobie/protoc-gen-php/blob/master/proto/php_options.proto

So, Basically we don't need to use extension for internal use. 
it is useful feature if you provide proto file to another developer.
```
